proxy/proxy.go

```diff
@@ -283,6 +297,12 @@ func WithParams(p Params) *Proxy {
 		m = metrics.Void
 	}
 
+	if p.MaxLoopbacks == 0 {
+		p.MaxLoopbacks = DefaultMaxLoopbacks
+	} else if p.MaxLoopbacks < 0 {
+		p.MaxLoopbacks = 0
+	}
```

Comment:
Is this a common pattern in go?
```
if foo = 0
  foo = DEFAULT
else if foo < 0
  foo = 0
```

To me, this is an evil and confusing pattern